core/services/gateway/handlers/functions/subscriptions_test.go

```diff
@@ -57,10 +56,55 @@ func TestSubscriptions(t *testing.T) {
 		assert.NoError(t, subscriptions.Close())
 	})
 
+	// initially we have 3 subs and range is 3, which needs one pass
 	gomega.NewGomegaWithT(t).Eventually(func() bool {
 		expectedBalance := big.NewInt(0).SetBytes(hexutil.MustDecode("0x01158e460913d00000"))
 		balance, err1 := subscriptions.GetMaxUserBalance(common.HexToAddress(validUser))
 		_, err2 := subscriptions.GetMaxUserBalance(common.HexToAddress(invalidUser))
 		return err1 == nil && err2 != nil && balance.Cmp(expectedBalance) == 0
 	}, testutils.WaitTimeout(t), time.Second).Should(gomega.BeTrue())
 }
+
+func TestSubscriptions_MultiPass(t *testing.T) {
+	const ncycles int32 = 5
+	var currentCycle atomic.Int32
+	getSubscriptionCount := hexutil.MustDecode("0x0000000000000000000000000000000000000000000000000000000000000006")
+	getSubscriptionsInRange := hexutil.MustDecode("0x00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000003000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000001600000000000000000000000000000000000000000000000000000000000000240000000000000000000000000000000000000000000000000de0b6b3a76400000000000000000000000000000109e6e1b12098cc8f3a1e9719a817ec53ab9b35c000000000000000000000000000000000000000000000000000034e23f515cb0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000c000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000f5340f0968ee8b7dfd97e3327a6139273cc2c4fa000000000000000000000000000000000000000000000001158e460913d000000000000000000000000000009ed925d8206a4f88a2f643b28b3035b315753cd60000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000c0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001bc14b92364c75e20000000000000000000000009ed925d8206a4f88a2f643b28b3035b315753cd60000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000c0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000010000000000000000000000005439e5881a529f3ccbffc0e82d49f9db3950aefe")
```

Comment:
Is it possible to encode this data within the test to make it more readable?

Reply:
This would require to load ABI, etc. So it should be possible, but probably not that easy and it will make the test more complicated.